lib/update/cluster/plan.go

```diff
@@ -47,13 +47,20 @@ import (
 	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
 )
 
+// UserConfig holds configuration parameters provided by the user triggering the update operation.
+type UserConfig struct {
+	ParallelWorkers int
```

Comment:
Validation wise, we probably want this to be a positive integer.  Setting 0 ornegative integers are accepted, but silently coerced to serial execution. It doesn't break anything perse, just leaves undefined behavior in the UX.

Reply:
Ended up tweaking this in 22eb5a3b292090bd33fb5ba4a7a9a371132283f2

Although I did the validation in the CLI code, I think it's fine for the library code to treat 0 or 1 as serial execution.